src/Illuminate/Support/LazyCollection.php

```diff
@@ -0,0 +1,1402 @@
+     */
+    public function random($number = null)
+    {
+        $result = $this->collect()->random(...func_get_args());
+
+        return is_null($number) ? $result : new static($result);
+    }
+
+    /**
+     * Reduce the collection to a single value.
+     *
+     * @param  callable  $callback
+     * @param  mixed  $initial
+     * @return mixed
+     */
+    public function reduce(callable $callback, $initial = null)
+    {
+        $result = $initial;
+
+        foreach ($this as $value) {
+            $result = $callback($result, $value);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Replace the collection items with the given items.
+     *
+     * @param  mixed  $items
+     * @return static
+     */
+    public function replace($items)
+    {
+        return new static(function () use ($items) {
+            $items = $this->getArrayableItems($items);
+            $usedItems = [];
+
+            foreach ($this as $key => $value) {
+                if (array_key_exists($key, $items)) {
+                    yield $key => $items[$key];
+
+                    $usedItems[$key] = true;
+                } else {
+                    yield $key => $value;
+                }
+            }
+
+            foreach ($items as $key => $value) {
+                if (! array_key_exists($key, $usedItems)) {
+                    yield $key => $value;
+                }
+            }
+        });
+    }
+
+    /**
+     * Recursively replace the collection items with the given items.
+     *
+     * @param  mixed  $items
+     * @return static
+     */
+    public function replaceRecursive($items)
+    {
+        return $this->passthru('replaceRecursive', func_get_args());
+    }
+
+    /**
+     * Reverse items order.
+     *
+     * @return static
+     */
+    public function reverse()
+    {
+        return $this->passthru('reverse', func_get_args());
+    }
+
+    /**
+     * Search the collection for a given value and return the corresponding key if successful.
+     *
+     * @param  mixed  $value
+     * @param  bool  $strict
+     * @return mixed
+     */
+    public function search($value, $strict = false)
+    {
+        $predicate = $this->useAsCallable($value)
+            ? $value
+            : function ($item) use ($value, $strict) {
+                return $strict ? $item === $value : $item == $value;
+            };
+
+        foreach ($this as $key => $item) {
+            if ($predicate($item, $key)) {
+                return $key;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get and remove the first item from the collection.
+     *
+     * @return mixed
+     */
+    public function shift()
+    {
+        $value = $this->getIterator()->current();
+
+        $source = $this->source;
+
+        $this->source = function () use ($source) {
+            $iterator = $this->makeIterator($source);
+
+            $iterator->next();
+
+            yield from $iterator;
```

Comment:
This throws a "Generator passed to yield from was aborted without proper return and is unable to continue" exception:

```php
$c = new LazyCollection(['foo']);
$c->shift();
$c->shift();
```

This fixes it for me:

```suggestion
            yield from $iterator->valid() ? $iterator : [];
```

Reply:
Actually, I just realized that this is a way simpler implementation:

```php
return tap($this->first(), function () {
    $this->source = $this->skip(1);
});
```